docs/streaming/structured-streaming-state-data-source.md

```diff
@@ -185,6 +203,20 @@ These instances logically compose buffers to store the input rows for left and r
 Since it is more obvious to users to reason about, the data source provides the option 'joinSide' to read the buffered input for specific side of the join.
 To enable the functionality to read the internal state store instance directly, we also allow specifying the option 'storeName', with restriction that 'storeName' and 'joinSide' cannot be specified together.
 
+### Reading state for transformWithState
+
+TransformWithState is a stateful operator that allows users to maintain arbitrary state across batches. In order to read this state, the user needs to provide some additional options in the state data source reader query.
+This operator allows for multiple state variables to be used within the same query. However, because they could be of different composite types and encoding formats, they need to be read within a batch query one variable at a time.
+In order to allow this, the user needs to specify the `stateVarName` for the state variable they are interested in reading.
+
+Timers can read by setting the option `readRegisteredTimers` to true. This will return all the registered timer across grouping keys.
```

Comment:
nit: maybe, `You can read timers by`, or `Timers can be read by`?

Reply:
Done